src/Datadog.Trace/Agent/Api.cs

```diff
@@ -74,39 +62,64 @@ public async Task SendTracesAsync(Span[][] traces)
 
             while (true)
             {
-                HttpResponseMessage responseMessage;
+                var request = _apiRequestFactory.Create(_tracesEndpoint);
```

Comment:
Strictly speaking, we do not NEED to abstract calls to the API here. We could very well use `WebRequest.Create(Uri uri)` directly, but that was difficult to unit test / mock. Let me know if I should pull back on that and just strongly type against `HttpWebRequest` here

Reply:
Testability is good!